client/ayon_core/modules/deadline/plugins/publish/collect_default_deadline_server.py

```diff
@@ -33,15 +32,17 @@ def process(self, context):
         deadline_settings = context.data["project_settings"]["deadline"]
         deadline_server_name = deadline_settings["deadline_server"]
 
-        deadline_webservice = None
+        dl_ws_item = None
         if deadline_server_name:
-            deadline_webservice = deadline_module.deadline_urls.get(
+            dl_ws_item = deadline_module.deadline_server_info.get(
                 deadline_server_name)
 
-        default_deadline_webservice = deadline_module.deadline_urls["default"]
-        deadline_webservice = (
-            deadline_webservice
-            or default_deadline_webservice
-        )
+        if dl_ws_item:
+            deadline_url = dl_ws_item["value"]
+        else:
+            default_dl_item = deadline_module.deadline_server_info.pop()
```

Comment:
I'm pretty sure you want to `pop` from the static `deadline_server_info` on the module? Why are we removing it from there?